src/libs/HttpUtils.js

```diff
@@ -18,6 +19,15 @@ Onyx.connect({
     callback: val => shouldFailAllRequests = (val && _.isBoolean(val.shouldFailAllRequests)) ? val.shouldFailAllRequests : false,
 });
 
+// getEnvironment promise returns the environment based on ENVIRONMENT Config for web/desktop.
+// Additionally, for iOS/android betaChecker.isBetaBuild() does the version check
+// environment is initialized as null as the default set in CONFIG.js always takes precedence
+let environment = null;
+Environment.getEnvironment()
```

Comment:
Does this function read a different Config from what we have in env which was used to build the app?

Reply:
I still do not get it. Why getting the value here is different from Config.js file?

Reply:
@parasharrajat 

For web/desktop the function reads the config as you said in libs/Environment/getEnvironment/index.js.

```
function getEnvironment() {
    return Promise.resolve(lodashGet(Config, 'ENVIRONMENT', CONST.ENVIRONMENT.DEV));
}
```

For Native, the function in libs/Environment/getEnvironment/index.native.js additionally does a isBetaBuild() check. See below. The isBetaBuild compares the pkg.version with the Google Play Store version and accordingly sets isBeta flag. iOS does the same check using NativeModules.EnvironmentChecker check for iOS devices. 

```
if (lodashGet(Config, 'ENVIRONMENT', CONST.ENVIRONMENT.DEV) === CONST.ENVIRONMENT.DEV) {
      environment = CONST.ENVIRONMENT.DEV;
      return resolve(environment);
  }

// If we haven't set the environment yet and we aren't on dev, check to see if this is a beta build
betaChecker.isBetaBuild()
    .then((isBeta) => {
        environment = isBeta ? CONST.ENVIRONMENT.STAGING : CONST.ENVIRONMENT.PRODUCTION;
        resolve(environment);
    });
```

Reply:
@smrutiparida Thanks for the explanation, can you also include this in the PR body?

Reply:
Ok, I got it.